### ISSUE TITLE
Tortoise mutating webhook removes appArmorProfile from pod spec

### ISSUE THREAD

**Issue by randytqwjp** [mercari/tortoise#417]
Kubernetes v1.30 deprecates enabling appArmor through annotations and enables it via securitycontext. Tortoise seems to remove this securitycontext through the mutating webhook `{"configuration":"tortoise-mutating-webhook-configuration","webhook":"mpod.kb.io","patch":[{"op":"remove","path":"/spec/containers/0/securityContext/appArmorProfile"},`

**Comment by randytqwjp** [comment]
@sanposhiho I suspect this could be due to underlying libs like container runtime being out of date causing the field to be securitycontext field to be dropped. An upgrade to support kubernetes v1.30 would require a major version upgrade for kubebuilder https://github.com/kubernetes-sigs/kubebuilder/releases?q=v3&expanded=true

**Comment by sanposhiho** [comment]
Yeah, we don't touch security context in our webhooks directly ourselves, so it should be caused by underlying dependencies like kubebuilder.

**Comment by lchavey** [comment]
Will do next week.

**Comment by sanposhiho** [comment]
@randytqwjp so, was it solved by upgrading kubebuilder after all?
